stacks-signer/src/signer.rs

```diff
@@ -1059,25 +1052,26 @@ impl Signer {
 
     /// Process a dkg result by broadcasting a vote to the stacks node
     fn process_dkg(&mut self, stacks_client: &StacksClient, dkg_public_key: &Point) {
-        let mut dkg_results_bytes = vec![];
+        // Abort if an aggregate key has already been set for the current rewards cycle
+        if let Some(aggregate_key) = stacks_client
+            .get_approved_aggregate_key(self.reward_cycle)
+            .map_err(|e| error!("{self}: Unable to get approved aggregate key: {e}"))
+            .unwrap_or_default()
+        {
+            self.approved_aggregate_public_key = Some(aggregate_key);
```

Comment:
Again, I think you'll want to check that the aggregate key is consistent with the saved state?

Reply:
Yeah, I'll change `load_saved_state` to take the aggregate key as input to make the behavior more explicit here.

Reply:
Changed to only use the `load_saved_state_for_aggregate_key` method.